app/src/main/java/org/oppia/app/player/state/StateFragment.kt

```diff
@@ -5,12 +5,25 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.databinding.ObservableField
 import org.oppia.app.fragment.InjectableFragment
+import org.oppia.app.player.audio.CellularDataDialogFragment
+import org.oppia.app.player.audio.CellularDataInterface
 import javax.inject.Inject
 
-/** Fragment that represents the current state card of an exploration. */
-class StateFragment : InjectableFragment() {
-  @Inject lateinit var stateFragmentPresenter: StateFragmentPresenter
+private const val TAG_CELLULAR_DATA_DIALOG = "CELLULAR_DATA_DIALOG"
+
+/** Fragment that represents the current state of an exploration. */
+class StateFragment : InjectableFragment(), CellularDataInterface {
+  @Inject
+  lateinit var stateFragmentPresenter: StateFragmentPresenter
+  // Control this boolean value from controllers in domain module.
+  private var showCellularDataDialog = true
+  var isAudioFragmentShowing = ObservableField<Boolean>(false)
+
+  init {
+    // Add code to control the value of showCellularDataDialog using AudioController.
```

Comment:
Please frame this as a TODO pointing to the correct GitHub issue to make this easier to track.